components/ai_chat/core/browser/engine/engine_consumer_oai.cc

```diff
@@ -0,0 +1,234 @@
+           l10n_util::GetStringUTF8(IDS_AI_CHAT_CLAUDE_SYSTEM_MESSAGE_PART1),
+           {date_and_time_string}, nullptr)});
+
+  // build
+  base::Value::List messages;
+
+  // append system message
+  {
+    base::Value::Dict message;
+    message.Set("role", "system");
+    message.Set("content", system_message);
+    messages.Append(std::move(message));
+  }
+
+  for (const mojom::ConversationTurnPtr& turn : conversation_history) {
+    base::Value::Dict message;
+    message.Set("role", turn->character_type == CharacterType::HUMAN
+                            ? "user"
+                            : "assistant");
+    message.Set("content", turn->text);
+    messages.Append(std::move(message));
+  }
+
+  return messages;
+}
+
+}  // namespace
+
+EngineConsumerOAIRemote::EngineConsumerOAIRemote(
+    const mojom::CustomModel& model,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory) {
+  model_ = model;
+  max_page_content_length_ = 4000;
+  api_ = std::make_unique<OAIAPIClient>(url_loader_factory);
+}
+
+EngineConsumerOAIRemote::~EngineConsumerOAIRemote() = default;
+
+void EngineConsumerOAIRemote::ClearAllQueries() {
+  api_->ClearAllQueries();
+}
+
+bool EngineConsumerOAIRemote::SupportsDeltaTextResponses() const {
+  return true;
+}
+
+void EngineConsumerOAIRemote::GenerateRewriteSuggestion(
+    std::string text,
+    const std::string& question,
+    GenerationDataCallback received_callback,
+    GenerationCompletedCallback completed_callback) {
+  base::Value::List messages;
+  {
+    base::Value::Dict message;
+    message.Set("role", "system");
+    message.Set("content",
+                "This is an excerpt user selected to be rewritten. Put your "
+                "rewritten of the excerpt in <response></response> tags.");
+    messages.Append(std::move(message));
+  }
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "user");
+    message.Set("content", base::StrCat({text, question}));
+    messages.Append(std::move(message));
+  }
+
+  api_->PerformRequest(model_, std::move(messages),
+                       std::move(received_callback),
+                       std::move(completed_callback));
+}
+
+void EngineConsumerOAIRemote::GenerateQuestionSuggestions(
+    const bool& is_video,
+    const std::string& page_content,
+    SuggestedQuestionsCallback callback) {
+  const std::string& truncated_page_content =
+      page_content.substr(0, max_page_content_length_);
+  std::string content_segment = base::ReplaceStringPlaceholders(
+      l10n_util::GetStringUTF8(is_video
+                                   ? IDS_AI_CHAT_CLAUDE_VIDEO_PROMPT_SEGMENT
+                                   : IDS_AI_CHAT_CLAUDE_ARTICLE_PROMPT_SEGMENT),
+      {truncated_page_content}, nullptr);
+
+  base::Value::List messages;
+
+  {
+    base::Value::Dict message;
+    message.Set("role", "system");
+    message.Set("content",
+                "Separate each suggested question with the character |");
```

Comment:
@LorenzoMinto this is flaky in some models except open ai's gpts

Reply:
Let's switch to `<question>$Q</question>` output format, that should work consistently across models and be more reliable.

Reply:
switched to `<question>$Q</question>`